### DISCUSSION TITLE
Can't render stream actions

### DISCUSSION THREAD

**Discussion by NicosKaralis** [yippee-fun/phlex#499] [Help] [answered]
I'm trying to replace my `*.turbo_stream.erb` actions with phlex components

Currently this is the erb:

``` erb
<% if @redirect.present? %>
  <%= turbo_stream_action_tag 'redirect', url: @redirect %>
<% else %>
  <%= turbo_stream.replace 'new_session_form', render(Sessions::New::FormView.new(session: @session)) %>
<% end %>
```

And this is the class I've created to replace the erb

``` ruby
class Sessions::CreateStream < ApplicationView

  def initialize(redirect: nil)
    @redirect = redirect
  end

  def template
    if @redirect.present?
      unsafe_raw helpers.turbo_stream_action_tag 'redirect', url: @redirect
    else
      unsafe_raw helpers.turbo_stream.replace('new_session_form', inline: render(Sessions::New::FormView.new))
    end
  end
end
```

and in the controller I call like this:

```ruby
render Sessions::CreateStream.new, layout: false
```

This is the best result I've got, but it does not work

As you can see with the debug comments, the content of the `Sessions::New::FormView` are placed outside and before the template tag

```html
<!-- Before Sessions::CreateStream -->
<!-- Before Sessions::New::FormView -->
<turbo-frame id="new_session_form"><form > Form content</form></turbo-frame>
<!-- After Sessions::New::FormView -->
<turbo-stream action="replace" target="new_session_form"><template></template></turbo-stream>
<!-- After Sessions::CreateStream -->
```

What am I missing?

Other things I've tried:

```ruby
include Phlex::Rails::Helpers::TurboStream
and 
turbo_stream.replace('new_session_form', inline: render(Sessions::New::FormView.new))
```
This did not render anything for me

**Comment by joeldrapper** [comment]
Hey @NicosKaralis, which version of `turbo-rails` are you using?

The `turbo_stream` helper didn't support rendering _renderables_ (objects that respond to `render_in`, such as `Phlex::HTML` objects) until after this PR. https://github.com/hotwired/turbo-rails/pull/433

The PR has been merged, but hasn't been released yet. In the meantime, you may want to configure your Gemfile to use the latest edge version of `turbo-rails` like this:

```ruby
gem "turbo-rails", git: "https://github.com/hotwired/turbo-rails.git"
```

**Comment by joeldrapper** [comment, marked as answer]
Hey @NicosKaralis, my changes to turbo-rails were released in turbo-rails 1.4, so you can now use it like this:

```ruby
turbo_stream.append "notifications", NotificationComponent.new(@notification)
```
